File: Design2p1.java
class MyQueue {

    /** Initialize your data structure here. */
    Stack<Integer> stack1 ;
    Stack<Integer> stack2;
    public MyQueue() {
        stack1 = new Stack<Integer>();
        stack2 = new Stack<Integer>();
    }
    
    /** Push element x to the back of queue. */
    public void push(int x) {
        
        stack1.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    public int pop() {
       if(!stack2.isEmpty())
           return stack2.pop();
        
        else
        {
            while(!stack1.isEmpty()){
                stack2.push(stack1.pop());
            }
        }
        return stack2.pop();
        
          
    }
    
    /** Get the front element. */
    public int peek() {
         if(!stack2.isEmpty())
           return stack2.peek();
        
        else
        {
            while(!stack1.isEmpty()){
                stack2.push(stack1.pop());
            }
        }
        return stack2.peek();
    }
    /** Returns whether the queue is empty. */
    public boolean empty() {
        if(stack1.isEmpty() && stack2.isEmpty())
            return true;
        else
            return false;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */


   

File: Design-2-P-1.py
class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stack = []
        self.popstack = []
        self.front = 0

    def push(self, x) :
        """
        Push element x to the back of queue.
        """
        if len(self.stack) == 0:
            self.front = x
        self.stack.append(x)

    def pop(self) :
        """
        Removes the element from in front of queue and returns that element.
        """
        if len(self.popstack) == 0:
            while len(self.stack) != 0:
                self.popstack.append(self.stack.pop())
        self.popstack.pop()

    def peek(self):
        """
        Get the front element.
        """
        if len(self.popstack) != 0:
            print self.popstack.peek()
        print self.front

    def empty(self) :
        """
        Returns whether the queue is empty.
        """
        if len(self.stack) == 0 and len(self.popstack) == 0:
            print ("stack is empty")
        else :
            print("stack is not empty")

# Your MyQueue object will be instantiated and called as such:
obj=MyQueue()
obj.push(10)
obj.push(20)
obj.pop()
obj.push(30)
obj.push(40)
obj.peek()
obj.empty()